# Copyright (c) 2024, Personal Robotics Laboratory
# License: BSD 3-Clause. See LICENSE.md file in root directory.

# MoveIt uses this configuration for controller management

moveit_controller_manager: moveit_ros_control_interface/Ros2ControlManager

moveit_simple_controller_manager:
  controller_names:
    - jaco_arm_controller
    - jaco_arm_cartesian_controller
    - jaco_arm_servo_controller
    - hand_controller

  jaco_arm_controller:
    type: FollowJointTrajectory
    action_ns: follow_joint_trajectory
    default: true
    joints:
      - j2n6s200_joint_1
      - j2n6s200_joint_2
      - j2n6s200_joint_3
      - j2n6s200_joint_4
      - j2n6s200_joint_5
      - j2n6s200_joint_6

  jaco_arm_cartesian_controller:
    type: ""
    action_ns: cart_commands
    default: false
    joints:
      - j2n6s200_joint_1
      - j2n6s200_joint_2
      - j2n6s200_joint_3
      - j2n6s200_joint_4
      - j2n6s200_joint_5
      - j2n6s200_joint_6

  jaco_arm_servo_controller:
    type: ""
    action_ns: commands
    default: false
    joints:
      - j2n6s200_joint_1
      - j2n6s200_joint_2
      - j2n6s200_joint_3
      - j2n6s200_joint_4
      - j2n6s200_joint_5
      - j2n6s200_joint_6
  hand_controller:
    type: FollowJointTrajectory
    action_ns: follow_joint_trajectory
    default: true
    joints:
      - j2n6s200_joint_finger_1
      - j2n6s200_joint_finger_2
